File: app/src/main/java/com/example/test/UserListFragment.java
package com.example.test;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProvider;

import com.example.test.databinding.FragmentUserListBinding;
import com.example.test.model.adapter.UserListAdapter;
import com.example.test.model.adapter.UserListLoadStateAdapter;
import com.example.test.model.fragment.ViewBindingFragment;
import com.example.test.viewmodels.UserListViewModel;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class UserListFragment extends ViewBindingFragment<FragmentUserListBinding> {

    private UserListAdapter adapter;

    @NonNull
    @Override
    protected FragmentUserListBinding inflateBinding(@NonNull LayoutInflater inflater, @Nullable ViewGroup container) {
        return FragmentUserListBinding.inflate(inflater, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        initRecyclerviewAndAdapter();
        UserListViewModel viewModel = new ViewModelProvider(this).get(UserListViewModel.class);
        viewModel.userDataFlowable.subscribe(moviePagingData -> adapter.submitData(getLifecycle(), moviePagingData));
    }

    private void initRecyclerviewAndAdapter() {
        adapter = new UserListAdapter();
        getBinding().userList.setAdapter(
                adapter.withLoadStateFooter(new UserListLoadStateAdapter())
        );
    }
}

File: app/src/main/java/com/example/test/UserListFragment.kt
package com.example.test

import android.os.*
import android.view.*
import androidx.fragment.app.*
import androidx.lifecycle.lifecycleScope
import com.example.test.adapters.*
import com.example.test.databinding.FragmentUserViewBinding
import com.example.test.viewmodels.*
import dagger.hilt.android.*
import kotlinx.coroutines.Job
import kotlinx.coroutines.flow.collectLatest
import kotlinx.coroutines.launch

@AndroidEntryPoint
class UserListFragment : Fragment() {

    private var binding: FragmentUserViewBinding? = null
    private var searchJob: Job? = null
    private val adapter = UsersAdapter()
    private val viewModel: UserListViewModel by viewModels()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        binding = FragmentUserViewBinding.inflate(inflater, container, false)
        return binding!!.root
    }

    override fun onDestroyView() {
        searchJob?.cancel()
        binding = null
        super.onDestroyView()
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding!!.rvUserList.adapter = adapter
        subscribeUi()
    }

    private fun subscribeUi() {
        searchJob?.cancel()
        searchJob = lifecycleScope.launch {
            viewModel.getResult().collectLatest {
                adapter.submitData(it)
            }
        }
    }
}